kornia/core/external.py

```diff
@@ -0,0 +1,28 @@
+import importlib
+
+
+class LazyLoader:
+    def __init__(self, module_name):
+        self.module_name = module_name
+        self.module = None
+
+    def _load(self):
+        if self.module is None:
+            try:
+                self.module = importlib.import_module(self.module_name)
+            except ImportError as e:
+                raise ImportError(
+                    f"Optional dependency '{self.module_name}' is not installed. "
+                    f"Please install it to use this functionality."
+                ) from e
+
+    def __getattr__(self, item):
+        self._load()
+        return getattr(self.module, item)
+
+    def __dir__(self):
+        self._load()
+        return dir(self.module)
+
+
+diffusers = LazyLoader("diffusers")
```

Comment:
is it needed to register this lib ? or can be done directly where is reallty used, e.g in StableDiffusionDissolving

Reply:
It is better to register here I think. Otherwise, we may have multiple singleton class towards the same lib.